CRToast/CRToastLayoutHelpers.h

```diff
@@ -205,13 +205,17 @@ static CGRect CRGetNotificationContainerFrame(UIInterfaceOrientation statusBarOr
 
 /// Get view snapshot. If `underStatusBar` it will get key windows root view controller. Otherwise it'll get the mainscreens snapshot
 static UIView *CRStatusBarSnapShotView(BOOL underStatusBar) {
-    if([[[UIDevice currentDevice] systemVersion] floatValue] >= 8.0) {
-        return underStatusBar ?
-        [[UIApplication sharedApplication].keyWindow.rootViewController.view snapshotViewAfterScreenUpdates:YES] :
-        [[UIScreen mainScreen] snapshotViewAfterScreenUpdates:YES];
-    } else {
-        return underStatusBar ?
-        [[UIApplication sharedApplication].keyWindow.rootViewController.view snapshotViewAfterScreenUpdates:NO] :
-        [[UIScreen mainScreen] snapshotViewAfterScreenUpdates:NO];
-    }
-}
\ No newline at end of file
+	__block UIView *snapshot;
+	dispatch_async(dispatch_get_main_queue(), ^(void){
+		if([[[UIDevice currentDevice] systemVersion] floatValue] >= 8.0) {
+			snapshot = underStatusBar ?
+			[[UIApplication sharedApplication].keyWindow.rootViewController.view snapshotViewAfterScreenUpdates:YES] :
+			[[UIScreen mainScreen] snapshotViewAfterScreenUpdates:YES];
+		} else {
+			snapshot = underStatusBar ?
+			[[UIApplication sharedApplication].keyWindow.rootViewController.view snapshotViewAfterScreenUpdates:NO] :
+			[[UIScreen mainScreen] snapshotViewAfterScreenUpdates:NO];
+		}
+	});
+    return snapshot;
```

Comment:
Agreed. This was the result of the quick cleaning. We got pretty panicked with the issue :)